source/extensions/transport_sockets/tls/ssl_handshaker.cc

```diff
@@ -98,6 +98,12 @@ Network::PostIoAction SslHandshakerImpl::doHandshake() {
     case SSL_ERROR_WANT_CERTIFICATE_VERIFY:
       state_ = Ssl::SocketState::HandshakeInProgress;
       return PostIoAction::KeepOpen;
+    case SSL_ERROR_SYSCALL:
```

Comment:
Is this change needed for error handling to correctly increment `was_key_usage_invalid_`? Or is it an unrelated change?